meson.build

```diff
@@ -29,29 +29,35 @@ else
     endif
 endif
 
-curl_dep = dependency('libcurl')
+curl_dep = dependency('curl', modules: 'CURL::libcurl')
 
-if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
-    argplib = cc.find_library('argp')
-else
-    argplib = dependency('', required : false)
+if build_machine.system() != 'windows'
+    if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
+       argplib = cc.find_library('argp')
```

Comment:
I don't have argp for Windows, but I think even with argp it might be challenging to build the tests / executables.

Reply:
I think for Windows if you don't want to build the programs, just make it a https://mesonbuild.com/Reference-manual_functions.html#disabler

It will cause any build targets that have the disabler as a dependency, to be no-op'ed and be converted into a disabler() in turn.

Reply:
@eli-schwartz, could you please give an example of how we could use a disabler here?  If it would simplify `meson.build`, I think that's the road I'd like to go down.

Reply:
@wolfv, is there truly no argp for windows?  Nothing that we can get out of musl?

Reply:
```diff
diff --git a/meson.build b/meson.build
index 0600fa4..8a818e7 100644
--- a/meson.build
+++ b/meson.build
@@ -39,8 +35,11 @@ else
     curl_dep = dependency('libcurl')
 endif
 
-if build_machine.system() != 'windows'
-    if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
+if build_machine.system() == 'windows'
+    argplib = disabler()
+else
+    # in meson 0.49, if build_machine.system() in ['freebsd', 'darwin']
+    if ['freebsd', 'darwin'].contains(build_machine.system()) or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}', name: 'argp_usage test')
        argplib = cc.find_library('argp')
     else
        argplib = dependency('', required : false)
```

You can just make it optional, too:

```diff
diff --git a/meson.build b/meson.build
index 0600fa4..8a818e7 100644
--- a/meson.build
+++ b/meson.build
@@ -39,12 +35,16 @@ else
     curl_dep = dependency('libcurl')
 endif
 
-if build_machine.system() != 'windows'
-    if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
-       argplib = cc.find_library('argp')
-    else
-       argplib = dependency('', required : false)
-    endif
+# in meson 0.49, if build_machine.system() in ['freebsd', 'darwin']
+if ['freebsd', 'darwin'].contains(build_machine.system()) or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}', name: 'argp_usage test')
+   argplib = cc.find_library('argp', required: false)
+   if not argplib.found()
+       argplib = disabler()
+   endif
+   # in meson 0.49, do:
+   # argplib = cc.find_library('argp', required: false, disabler: true)
+else
+   argplib = dependency('', required : false)
 endif
 
 if get_option('with-openssl') == 'disabled'
```

Reply:
I have an argp with necessary fixes for Windows. I'll try to polish it enough to make it useful for our usage here.